progress_go1.7.go

```diff
@@ -0,0 +1,16 @@
+//+build go1.7
+
+package mpb
+
+import "context"
+
+// New Progress instance, it orchestrates the rendering of progress bars.
+// It supports cancellation via Context.
+func NewWithCtx(ctx context.Context) *Progress {
+	p := New()
+	go func() {
+		<-ctx.Done()
+		p.Stop()
+	}()
+	return p
+}
```

Comment:
I am not sure how useful is it to provide this functionality. If someone requires it they could easily implement it like that. Any thoughts?

My motivation for isolating context usage down to here is to make the library compatible with older versions of Go.

Having to do New(nil) also feels a bit weird, hence the separate constructor.